### ISSUE TITLE
False positive - RedundantModifier: Redundant 'public' modifier with nested public class

### ISSUE THREAD

**Issue by hgschmie** [checkstyle/checkstyle#13058]
Tested with 10.8.0, 10.10.0, 10.11.0

```bash
/var/tmp $ javac YOUR_FILE.java

/var/tmp $ cat config.xml

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="RedundantModifier"/>
    </module>
</module>
```

/var/tmp $ cat YOUR_FILE.java
```java
package checkstyle;

class Checkstyle {
    public static class Bar {
        private final int x;

        public Bar(int x) {
            this.x = x;
        }
    }
}
```

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml YOUR_FILE.java

```
❯ java -jar ~/Downloads/checkstyle-10.11.0-all.jar -c ./checkstyle.xml Checkstyle.java
Starting audit...
[ERROR] /private/tmp/Checkstyle.java:7:9: Redundant 'public' modifier. [RedundantModifier]
Audit done.
Checkstyle ends with 1 errors.
```

The error is incorrect. The nested class is public so code that needs to inspect it (or call the constructor) can do so without having to call `setAccessible`. This is a very common pattern in unit tests (where the main test class is package private but the nested classes are public so they can be inspected.

checkstyle reports those as redundant; they are not. Removing the `public` modifier from the c'tor makes tests that need to introspect the constructor or call it from outside the package fail.

Making the outer class public is a workaround.

**Comment by hgschmie** [comment]
May be a duplicate of #12216 which is now open for almost a year.
